generator/src/Command/ScrapeCommand.php

```diff
@@ -0,0 +1,235 @@
+            // Create a new document for each row that starts with a non-empty cell
+            if ($row[0] && $docs !== []) {
+                $docId++;
+                $argId = 0;
+            }
+
+            foreach ($row as $index => $cell) {
+                if (str_contains($headers[$index], 'Arg')) {
+                    $docs[$docId]['Args'][$argId][str_replace('Arg', '', $headers[$index])] = $cell;
+                } elseif (null !== $cell) {
+                    $docs[$docId][$headers[$index]] = $cell;
+                }
+            }
+
+            $argId++;
+        }
+
+        return $docs;
+    }
+
+    /**
+     * @param array{
+     *            Name: string,
+     *            Category: string,
+     *            Description: string,
+     *            Link: string,
+     *            ReturnType: string,
+     *            Encode: string,
+     *            Args: array{ Name: string, Type: string, Options: string, Description: string }
+     *        } $doc
+     */
+    private function formatSpec(array $doc): array
+    {
+        foreach (['Name', 'Category', 'Description', 'Link', 'ReturnType', 'Encode', 'Args'] as $key) {
+            assert(isset($doc[$key]), 'Missing ' . $key . ' for ' . var_export($doc, true));
+        }
+
+        $spec = [];
+        $spec['name'] = $doc['Name'];
+        $spec['category'] = explode(PHP_EOL, $doc['Category']);
+        sort($spec['category']);
+        $spec['link'] = $doc['Link'];
+        $spec['returnType'] = explode(PHP_EOL, $doc['ReturnType']);
+        $spec['encode'] = $doc['Encode'];
+
+        if ($doc['Description']) {
+            $spec['description'] = $doc['Description'] . PHP_EOL;
+        }
+
+        foreach ($doc['Args'] as $arg) {
+            foreach (['Name', 'Type', 'Options', 'Description'] as $key) {
+                assert(isset($arg[$key]), 'Missing Arg' . $key . ' for ' . var_export($doc, true));
+            }
+
+            $parameter = [];
+            $parameter['name'] = $arg['Name'];
+            $parameter['type'] = explode(PHP_EOL, $doc['ReturnType']);
+            if (str_contains($arg['Options'], 'Optional')) {
+                $parameter['optional'] = true;
+            }
+
+            if ($arg['Description']) {
+                $parameter['description'] = $arg['Description'] . PHP_EOL;
+            }
+
+            $spec['parameters'][] = $parameter;
+        }
+
+        return $spec;
+    }
+
+    private function writeYamlFile(string $dirname, array $data): void
+    {
+        $yaml = Yaml::dump($data, 3, 4, Yaml::DUMP_MULTI_LINE_LITERAL_BLOCK);
+        $dirname = $this->configDir . '/' . $dirname;
+        if (! file_exists($dirname)) {
+            mkdir($dirname, 0755);
+        }
+
+        $name = str_replace('$', '', $data['name']) ?: 'positional';
+        $filename = $dirname . '/' . $name . '.yaml';
+
+        // Add a schema reference to the top of the file
+        $schema = '# $schema: ../schema.json' . PHP_EOL;
```

Comment:
PHPStorm will [support](https://youtrack.jetbrains.com/issue/IDEA-323117) it soon.

> Will be available in 2023.3 Release.